File: Dominio/ObjetosDeValor/Embarcador/Enumeradores/SituacaoGestaoDevolucao.cs
namespace Dominio.ObjetosDeValor.Embarcador.Enumeradores
{
    public enum SituacaoGestaoDevolucao
    {
        Ativa = 0,
        AnaliseCancelamento = 1,
        Cancelada = 2,
        Finalizada = 3,
    }

    public static class SituacaosGestaoDevolucaoHelper
    {
        public static string ObterDescricao(this SituacaoGestaoDevolucao situacaoGestaoDevolucao)
        {
            switch (situacaoGestaoDevolucao)
            {
                case SituacaoGestaoDevolucao.Ativa: return "Ativa";
                case SituacaoGestaoDevolucao.AnaliseCancelamento: return "Análise de Cancelamento";
                case SituacaoGestaoDevolucao.Cancelada: return "Cancelada";
                case SituacaoGestaoDevolucao.Finalizada: return "Finalizada";
                default: return string.Empty;
            }
        }
    }
}

File: SGT.WebAdmin/ViewsScripts/Enumeradores/SituacaoGestaoDevolucao.js
var EnumSituacaoGestaoDevolucaoHelper = function () {
    this.Ativa = 0;
    this.AnaliseCancelamento = 1;
    this.Cancelada = 2;
    this.Finalizada = 3;
};

EnumSituacaoGestaoDevolucaoHelper.prototype = {
    obterOpcoes: function () {
        return [
            { text: "Ativa", value: this.Ativa },
            { text: "Análise de Cancelamento", value: this.AnaliseCancelamento },
            { text: "Cancelada", value: this.Cancelada },
            { text: "Finalizada", value: this.Finalizada }
        ];
    },
    obterOpcoesPesquisa: function () {
        return [
            { text: "Ativa", value: this.Ativa },
            { text: "Análise de Cancelamento", value: this.AnaliseCancelamento },
            { text: "Cancelada", value: this.Cancelada },
            { text: "Finalizada", value: this.Finalizada }
        ];
    }
};

var EnumSituacaoGestaoDevolucao = Object.freeze(new EnumSituacaoGestaoDevolucaoHelper());